pkg/config/config.go

```diff
@@ -596,3 +609,14 @@ func (b *Blueprint) DeploymentName() (string, error) {
 	}
 	return deploymentName, nil
 }
+
+// checkBlueprintName returns an error if blueprint_name does not comply with
+// requirements for correct label values as per https://cloud.google.com/resource-manager/docs/creating-managing-labels#requirements
+func (b *Blueprint) checkBlueprintName() error {
```

Comment:
Based on what Carlos suggested, I'd change this function to `checkLabelValue` with the same validation. With that, you could run the same test against deployment_name in the Config.DeploymentName() function as well.

This also helps generalize as we add features. There may be other labels that we track and having the function ready to go will be very valuable.